### PR TITLE
docs: Add missing columns to worker docs

### PR DESCRIPTION
**Pull request by mrnugget**
I couldn't get a new worker to run without these three columns, so
adding them here.

### CODE REVIEW

File: doc/dev/background-information/workers.md [hunk 1]
```diff
@@ -84,17 +84,19 @@ The most common way to use a worker is to use the database-backed store. When us
...
+| `failure_message`   | text                     | Updated with the text of the error returned from the handle hook |
+| `started_at`        | timestamp with time zone | Updated when the job is dequeued for processing |
+| `finished_at`       | timestamp with time zone | Updated when the handler finishes processing the job (successfully or unsuccessfully) |
+| `process_after`     | timestamp with time zone | Controls the time after which the job is visible for processing |
+| `num_resets`        | integer                  | Updated when the job is moved back from `failed` to `queued` |
+| `num_failures`      | integer                  | Updated when the job enters the `failed` state |
+| `last_heartbeat_at` | timestamp with time zone | Updated periodically to ensure that the handler didn't die processing the job |
+| `execution_logs`    | json[]                   | A list of log entries from the most recent processing attempt |
+| `worker_hostname`   | text                     | Hostname of the `executor` that picked up the job. Used only when using executor. |
```
Comment (eseliger) [inline doc/dev/background-information/workers.md:99]: Is that true? I think it should also store the hostname for regular dbworkers🤔 or at least that's how I understood that field

**Review by eseliger — APPROVED** [review]

File: doc/dev/background-information/workers.md (hunk 1 above)
Comment (mrnugget) [reply doc/dev/background-information/workers.md:99]: No idea to be honest :grimacing: I'll take a look and change it.

Comment (mrnugget) [reply doc/dev/background-information/workers.md]: Yup, seems like it's updated by every worker.

Comment (efritz) [reply doc/dev/background-information/workers.md]: cc @Strum355

File: doc/dev/background-information/workers.md [hunk 2]
```diff
@@ -241,6 +254,10 @@ func scanFirstExampleJob(rows *sql.Rows, queryErr error) (_ workerutil.Record, e
 			return nil, false, err
 		}
 
+		for _, entry := range executionLogs {
```
Comment (efritz) [inline doc/dev/background-information/workers.md:257]: Never scanned?

Comment (mrnugget) [reply doc/dev/background-information/workers.md:260]: Fixed!

**Review by efritz — APPROVED** [review]
<3 thanks

File: doc/dev/background-information/workers.md (hunk 1 above)
Comment (Strum355) [reply doc/dev/background-information/workers.md]: that may have been a mistake on my end :thinking: dont remember context anymore

Comment (efritz) [reply doc/dev/background-information/workers.md]: Just wanted to ensure that worker_hostname is expected to be set everywhere (which I think is true).

Comment (Strum355) [reply doc/dev/background-information/workers.md]: The default/fallback is os.Hostname iirc, so it will always be set

**Pull request merged**